src/view/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,100 @@
+import { getApp } from "firebase/app";
+import {
+    getDownloadURL,
+    getStorage,
+    ref,
+    uploadBytesResumable,
+    UploadTask,
+} from "firebase/storage";
+import { useState } from "react";
+
+enum UploadRequestStatus {
+    IDLE = "IDLE",
+    PENDING = "PENDING",
+    FULFILLED = "FULFILLED",
+    REJECTED = "REJECTED",
+}
+
+const useUploadImage = () => {
+    const [files, setFiles] = useState<File[]>([]);
+    const [imageURLs, setImageURLs] = useState<string[]>([]);
+    const [uploadProgress, setUploadProgress] = useState<number | null>(null);
+    const [uploadRequestStatus, setUploadRequestStatus] =
+        useState<UploadRequestStatus>(UploadRequestStatus.IDLE);
+
+    const handleFileChange = (selectedFiles: FileList) => {
+        const selectedFilesArray = Array.from(selectedFiles);
+        setFiles(selectedFilesArray);
+        const imageURLsArray = selectedFilesArray.map((file) =>
+            URL.createObjectURL(file)
+        );
+        setImageURLs(imageURLsArray);
+    };
+
+    const handleUpload = async () => {
+        try {
+            setUploadRequestStatus(UploadRequestStatus.PENDING);
+            const firebaseApp = getApp();
+            const storage = getStorage(
+                firebaseApp,
+                process.env.CLOUD_STORAGE_POROTO_PLACE_IMAGES
+            );
+
+            const uploadTasks = files.map((file) => {
+                const storageRef = ref(storage, `images/${file.name}`);
+                return uploadBytesResumable(storageRef, file);
+            });
+
+            await Promise.all(uploadTasks.map(setupUploadTaskListener));
```

Comment:
個別のタスクの中でstatusの更新処理をすると、
一番はやくに終わったものが成功したら成功判定になっちゃうから
全部をまってあげたほうが良さそう。
```suggestion
            await Promise.all(uploadTasks.map(setupUploadTaskListener));
            setUploadRequestStatus(UploadRequestStatus.FULFILLED);
```